### ISSUE TITLE
Error "stream.push() after EOF" after ssh2 upgraded to 0.4.x

### ISSUE THREAD

**Issue by elleuca** [chuckmo/grunt-ssh#91]
Hi, I'm experiencing some "Fatal error: stream.push() after EOF" errors on a (previously) working project when I run `npm install` from scratch. 

Basically I've something like 

```
 grunt.config('sshconfig', {
        mycontainer: grunt.file.readJSON('mycontainer.json')
    });

    grunt.config('sshexec', {
        "check_connection": {
            command: 'uptime',
            options: {
                config: 'mycontainer'
            }
        },
  ...
```

Running `grunt --verbose sshexec:check_connection` returns now
```
Running "sshexec:check_connection" (sshexec) task
Verifying property sshexec.check_connection exists in config...OK
File: [no files]
Options: config="mycontainer", host=false, username=false, password=false, agent="", agentForward=false, port=22, ignoreErrors=false, minimatch={}, pty={}, suppressRemoteErrors=false
Raw Options: config="mycontainer", host=false, username=false, password=false, agent="", agentForward=false, port=22, ignoreErrors=false, minimatch={}, pty={}, suppressRemoteErrors=false
Verifying property sshconfig.mycontainer exists in config...OK
Options: config="mycontainer", host="ferretti.dev.loc", username="ferretti", password="password", agent="", agentForward=false, port=22, ignoreErrors=false, minimatch={}, pty={}, suppressRemoteErrors=false, path="/home/ferretti/projects/site"
Connection :: connect
Connection :: ready
Executing :: uptime
 12:48:49 up 5 days,  3:01,  2 users,  load average: 0.43, 1.71, 2.02
Stream :: exit :: code: 0, signal: undefined
Fatal error: stream.push() after EOF
```

The only difference I can see between now and friday is the upgrade of ssh2 from [0.3.x to 0.4.x](https://github.com/mscdex/ssh2/wiki/Changes-from-0.3.x-to-0.4.x).

Any ideas?

**Comment by sejoker** [comment]
I experience the same error, we should freeze ssh2 to 0.3.x until 0.4.x support will be tested.

**Comment by misshula** [comment]
+1

**Comment by bostrom** [comment]
Any workaround for this in the meantime?

**Comment by benholiio** [comment]
+1

**Comment by chrisse27** [comment]
+1

**Comment by TedSanne** [comment]
+1

**Comment by Mech7** [comment]
yes plz fix this

**Comment by andrewrjones** [comment]
Fixed in 0.12.1.
